src/preferences/dialog/dlgpreflv2.cpp

```diff
@@ -117,11 +117,15 @@ void DlgPrefLV2::slotApply() {
             m_pLV2Backend->getManifest(m_currentEffectId);
 
     qDebug() << "DlgPrefLV2::slotApply" << pCurrentEffectManifest.data();
+    unsigned int position = 0;
     for (int i = 0; i < m_pluginParameters.size(); i++) {
-        EffectManifestParameterPointer pParameter = pCurrentEffectManifest->parameter(i);
-        pParameter->setShowInParameterSlot(m_pluginParameters[i]->isChecked());
+        if (m_pluginParameters[i]->isChecked()) {
+            m_pEffectsManager->setEffectParameterPosition(pCurrentEffectManifest, i, position);
```

Comment:
Unfortunately this only changes effects that are currently loaded. If the LV2 plugin is loaded again later, all its parameters are loaded. As a quick solution we may consider going back to the ugly hack of setting a flag on the EffectManifestParameter and checking that flag in EffectSlot::loadEffect. In the long term I think we should implement [custom default EffectPresets for each effect type](https://bugs.launchpad.net/mixxx/+bug/1740504), but of course that requires having EffectPresets implemented first.

Reply:
Yes, I've added a TODO to remove these functions once the GUI is implemented for this. Once we have a GUI for hiding effect parameters, We'll be able to hide parameters for particular instances of EffectSlots.